spec/velveteen/worker_spec.rb

```diff
@@ -73,11 +87,19 @@ class TestSchemaWorker < described_class
   class TestPublishingWorker < described_class
     attr_accessor :test_headers
 
-    def perform
+    def perform(override_message = nil, publishing_key = "velveteen.test.publish")
+      if override_message
+        publish_message = override_message
+        publish_headers = override_message.headers
+      else
+        publish_message = message
+        publish_headers = test_headers || {}
+      end
```

Comment:
This was an effort handle the case where the message to publish is different or the same from the consumed message with defaults if the message to publish is the same as the message consumed. I was considering a method that "manipulates" the message but wanted to get an outside perspective on this.